MAUI/Autocomplete/Searching-Filtering.md

```diff
@@ -171,6 +171,276 @@ autocomplete.TextSearchMode = AutocompleteTextSearchMode.Contains;
+        if (filteredCapitals.Count() > 0)
+            return filteredCapitals.FirstOrDefault();
+
+        return 0;
+    }
+}
+
+{% endhighlight %}
+
+{% endtabs %}
+
+**Step 3:** Applying custom searching to `Autocomplete` control by using the `SearchBehavior` property. 
+
+{% tabs %}
+{% highlight XAML %}
+ <ContentPage.BindingContext>
+        <local:CityViewModel></local:CityViewModel>
+    </ContentPage.BindingContext>
+<editors:SfAutocomplete 
+    DisplayMemberPath="CityName"
+    ItemsSource="{Binding Cities}">
+            <editors:SfAutocomplete.FilterBehavior>
+                <local:CityFilteringBehavior/>
+            </editors:SfAutocomplete.FilterBehavior>
+<editors:SfAutocomplete.SearchBehavior>
+                <local:CapitalCitySearchingBehavior></local:CapitalCitySearchingBehavior>
+            </editors:SfAutocomplete.SearchBehavior>
+</editors:SfAutoComplete>
+
+{% endhighlight %}
+{% endtabs %}
+
+The following gif demonstrates how to select the capital city from the drop-down based on the country name entered in the `Autocomplete` control.
+
+![.NET MAUI AutoComplete filter the items based on custom filtering and searching logic.](Images/Searching_Filtering/Custom_filtering_searching.gif)
+
+## Load asynchronous items
+
+Load the data dynamically at runtime based on typed input. This dynamic loading can be done while performing custom filtering using the `CustomFilter` property.
+
+The `GetMatchingItemsAsync` method of `IAutocompleteFilterBehavior` helps you to perform filtering operations on different threads without blocking the current thread by using `await Task.Run()`.
+
+**Step 1:** Create a class that derives from the `IAutocompleteFilterBehavior` interface and add your custom filter logic in `GetMatchingItemsAsync` method to load the run time items based on typed input.
+
+{% tabs %}
+{% highlight C# %}
+
+public class CustomAsyncFilter : IAutocompleteFilterBehavior
+{
+    /// <summary>
+    /// Gets the cancellation token source.
+    /// </summary>
+    CancellationTokenSource cancellationTokenSource;
+
+    public async Task<object> GetMatchingItemsAsync(SfAutocomplete source, AutocompleteFilterInfo filterInfo)
+    {
+        if (this.cancellationTokenSource != null)
+        {
+            this.cancellationTokenSource.Cancel();
+            this.cancellationTokenSource.Dispose();
+        }
+
+        this.cancellationTokenSource = new CancellationTokenSource();
+        CancellationToken token = this.cancellationTokenSource.Token;
+
+        return await Task.Run(() =>
+        {
+            List<string> list = new List<string>();
+            for (int i = 0; i < 100000; i++)
+            {
+                list.Add(filterInfo.Text + i);
+            }
+
+            return list;
+        }, token);
+    }
+}
+
+{% endhighlight %}
+{% endtabs %}
+
+**Step 2:** Applying `CustomAsyncFilter` to `Autocomplete` control by using the `FilterBehavior` property. 
```

Comment:
**Step 2:** Apply the `CustomAsyncFilter` to the `Autocomplete` control by using the `FilterBehavior` property.